eng/common/pipelines/templates/stages/archetype-typespec-emitter.yml

```diff
@@ -0,0 +1,452 @@
+            exit 1
+          displayName: 'Unsupported package type'
+          condition: always()
+
+    - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+
+    - ${{ parameters.InitializationSteps }}
+
+    - task: PowerShell@2
+      displayName: 'Run initialize script'
+      inputs:
+        pwsh: true
+        filePath: ${{ parameters.EmitterPackagePath }}/eng/scripts/Initialize.ps1
+        arguments: -UseTypeSpecNext:$${{ parameters.UseTypeSpecNext }}
+        workingDirectory: ${{ parameters.EmitterPackagePath }}
+
+    - task: PowerShell@2
+      displayName: 'Run build script'
+      name: ci_build
+      inputs:
+        pwsh: true
+        filePath: ${{ parameters.EmitterPackagePath }}/eng/scripts/Build-Packages.ps1
+        arguments: >
+          -BuildNumber "$(Build.BuildNumber)"
+          -OutputDirectory "$(Build.ArtifactStagingDirectory)"
+          -PublishInternal:$${{ parameters.PublishInternal }}
+          -Prerelease:$${{ parameters.BuildPrereleaseVersion }}
+
+    - pwsh: |
+        $sourceBranch = '$(Build.SourceBranch)'
+        $buildReason = '$(Build.Reason)'
+        $buildNumber = '$(Build.BuildNumber)'
+
+        if ($buildReason -eq 'Schedule') {
+          $branchName = 'validate-typespec-scheduled'
+        } elseif ($sourceBranch -match "^refs/pull/(\d+)/(head|merge)$") {
+          $branchName = "validate-typespec-pr-$($Matches[1])"
+        } else {
+          $branchName = "validate-typespec-$buildNumber"
+        }
+
+        Write-Host "Setting variable 'branchName' to '$branchName'"
+        Write-Host "##vso[task.setvariable variable=branchName;isOutput=true]$branchName"
+      displayName: Set branch name
+      name: set_branch_name
+
+    - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
+      parameters:
+        artifactName: build_artifacts
+        artifactPath: $(Build.ArtifactStagingDirectory)
+
+# Publish stage
+# Responsible for publishing the packages in `build_artifacts/packages` and producing `emitter-package-lock.json`
+# Produces the artifact `publish_artifacts` which contains the following:
+#   emitter-package.json: Created using the package json from the build step. 
+#   emitter-package-lock.json: Created by calling `npm install` using `emitter-package.json`
+- ${{ if parameters.ShouldPublish }}:
+  - stage: Publish
+    dependsOn:
+    - Build
+    - ${{ if and(parameters.PublishDependsOnTest, ne(length(parameters.TestMatrix), 0)) }}:
+      - Test
+    variables:
+      buildArtifactsPath: $(Pipeline.Workspace)/build_artifacts
+    pool: ${{ parameters.Pool }}
+    jobs:
+    - job: Publish
+      steps:
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+
+      - download: current
+        artifact: build_artifacts
+        displayName: Download build artifacts
+
+      # Create authenticated .npmrc file for publishing
+      - ${{ if eq(parameters.PublishInternal, 'true') }}:
```

Comment:
Publish Internal is kind of misleading as it is going to a public DevOps Feed. You might want to flip this and make the called PublishPublicly or just call this PublishToDevOps or something similar.